src/java.desktop/macosx/classes/com/apple/laf/AquaTableHeaderBorder.java

```diff
@@ -105,7 +105,7 @@ public void paintBorder(final Component c, final Graphics g, final int x, final
         final int newWidth = width;
         final int newHeight = height;
 
-        painter.paint(g, c, newX - 1, newY - 1, newWidth + 1, newHeight);
+        painter.paint(g, c, newX - 1, newY - 1, newWidth + 1, newHeight + 1);
```

Comment:
I would like to clarify the change here, what is the coordinates/size passed here is it a size of the component+border, or just a size of the component? So if decrease the newX/Y will we draw in the component area, or accidentally draw outside?

Reply:
@mrserb   It is the size of component + border. The component being painted here is `DefaultTableCellHeaderRenderer`.

Tested under two cases
1. Decreasing x & y, keeping width & height same as above - Border gets drawn inside the component area making header look smaller.
2. Decreasing width & height, keeping x & y same - Border is not visible

Reply:
If it is a size of the component+border then how we can draw outside of it by decreasing the x/y?

Reply:
@mrserb Sorry, what I meant to convey is that the coordinate arguments sent to painter.paint() is including the component + border, and newX, newY, newHeight, newWidth are coordinates of component only. So when newX and newY are decreased by more than 1 in paint() method border seems to be drawn inside the component area and not outside